packages/backend/e2e/deleteExpense.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { test, expect } from '@playwright/test'
+import { STATUS_SUCCESS_200, STATUS_UNPROCESSABLE_ENTITY_422 } from '../src/models/statusCodes'
+import { connectToDb, db } from '../src/db'
+import {
+  expensesTable,
+  expenseCategoriesTable,
+  ExpensesTableRow,
+  expenseSubCategories,
+} from '../src/db/schema'
+import crypto from 'crypto'
+import { NOT_FOUND_ERROR } from '../src/models/errors/repositoryErrors'
+
+const BASE_URL = 'http://localhost:3000'
+
+test.describe('Delete Expense Endpoint', () => {
+  let fakeExpenseData1: ExpensesTableRow
+  let fakeExpenseId: string
+  let fakeCategory: any
+  let fakeSubCategory: any
+
+  test.beforeAll(async () => {
+    connectToDb()
+    const { expenseId, expenseData1, category, subCategory } = await assignFakeExpenseData()
+    fakeExpenseId = expenseId
+    fakeExpenseData1 = expenseData1
+    fakeCategory = category
+    fakeSubCategory = subCategory
+  })
+
+  test.describe('when required data fails validation', () => {
+    test('should return error message and 422 status code', async ({ request }) => {
+      // Missing or invalid expenseId
+      const response = await request.post(`${BASE_URL}/deleteexpense`, {
+        params: { expenseId: '' },
+      })
+      expect(response.status()).toBe(STATUS_UNPROCESSABLE_ENTITY_422)
+      const body = await response.json()
+      expect(body).toHaveProperty('error')
+    })
+  })
+  test.describe('when a non-existent expense is deleted', () => {
+    test('should return error message and 404 status code', async ({ request }) => {
+      // Simulate an internal server error by providing an invalid accountId
+      const response = await request.post(`${BASE_URL}/deleteexpense`, {
+        data: {
+          userId: '00000000-0000-0000-0000-000000000001',
+          accountId: crypto.randomUUID(),
+          expenseId: crypto.randomUUID(),
+        },
+      })
+      expect(response.status()).toBe(404)
+      const body = await response.json()
+      expect(body.error).toContain(NOT_FOUND_ERROR)
+    })
+  })
+
+  test.describe('when the expense is deleted successfully', () => {
+    test('should return the expense related to the expense id', async ({ request }) => {
+      // Insert test expenses directly into database
+      const [insertedExpense] = await db.insert(expensesTable).values(fakeExpenseData1).returning()
+
+      // Send DELETE request
+      const response = await request.post(`${BASE_URL}/deleteexpense`, {
+        data: {
+          userId: insertedExpense.userId,
+          accountId: insertedExpense.accountId,
+          expenseId: insertedExpense.id,
+        },
+      })
+
+      expect(response.status()).toBe(STATUS_SUCCESS_200)
+      const body = await response.json()
+      expect(body).toHaveProperty('expense')
```

Comment:
This isn't necessary since the next line tests that the field contains the expected object which also implicity tests for its existence